### DISCUSSION TITLE
PrepareCopyTask.java in release/3.7

### DISCUSSION THREAD

**Discussion by Asim-7** [TeamAmaze/AmazeFileManager#2947] [General]
Please add this to PrepareCopyTask.java in release/3.7 to avoid any pasting files & folders conflicts. 
This morning I saw this issue and made a fix that helped me alot. This is a temporary fix but for now it will help the users.

![Screenshot 2021-10-21 at 11 46 14](https://user-images.githubusercontent.com/25628828/138253754-f54fd42d-0101-44a1-8c2d-2aec7c1d12b8.png)

**Comment by VishalNehra** [comment]
Is this checking files with files and folders with folders? So, say, if I have xyz named directory in a given folder, I should ve allowed to paste a file with same name xyz?

**Reply by Asim-7** [reply]
**Yes** my solution checks file names with file names and folder names with folder names. 
**Yes** the XYZ file will be pasted because it not compared with directory name

**Fyi:** With default code right now I am only able to add **conflictingFiles** not **conflictingFolders**.  

Previously If I have **directory RTY** with **xyz file** and wanted to paste it into a **ABC directory** where **xyz file** is present ->
it should paste the **RTY directory** in **ABC directory** then total outcome in **ABC directory** will be: **RTY directory & xyz file**
**but**
it failed because it was comparing every destination file & folder with every paste file & folder (nice) but also comparing with files within paste directories (its wrong) i.e. **xyz file** in **ABC directory** was compared to **xyz file in RTY directory** (this was the issue)

I **solved it** by avoiding comparison to go into Paste subdirectories

**Comment by Asim-7** [comment]
**PrepareCopyTask.java in release/3.7**

Just now I have also fixed **file replacement issue**. The issue was: **dialogState == UNKNOWN** was not handled correctly due to which dialog actions for single files were effected when multiple files with the same name found in the same directory

![Screenshot 2021-10-21 at 15 42 45](https://user-images.githubusercontent.com/25628828/138290853-a45e34f4-eb93-4bb0-9158-c285314990ef.png)
.

**Comment by EmmanuelMess** [comment]
Please at least provide patches, not screenshots.
